### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by j1c**
Notable changes

- Choose among full svd, sparse svd, and randomized svd
- Remove python 3.4 support
- Docstrings for omni and MDS
- Refactor simulations
  - Remove weighted simulations. Same simulation function for binary and weighted sims. Default to binary simulation

### CODE REVIEW

File: graspy/embed/embed.py [hunk 1]
```diff
@@ -78,17 +105,16 @@ def fit(self, graph):
         # import graph(s) to an adjacency matrix using import_graph function
         # here
 
-        return self.lpm
+        return self
 
     def _fit_transform(self, graph):
         "Fits the model and returns the estimated latent positions"
-        try:
```
Comment (bdpedigo) [inline graspy/embed/embed.py]: If calling fit_transform with new data than the model was fit on, we want fit_transform to rerun fit regardless. Is that the purpose of removing the _check_is_fit? just making sure

File: graspy/embed/lse.py [hunk 2]
```diff
@@ -51,15 +75,19 @@ class LaplacianSpectralEmbed(BaseEmbed):
...
     """
 
-    def __init__(self, form='DAD', method=selectSVD, *args, **kwargs):
-        super().__init__(method=method, *args, **kwargs)
+    def __init__(self,
+                 form='DAD',
+                 n_components=None,
+                 n_elbows=2,
+                 algorithm='randomized',
+                 n_iter=5):
```
Comment (bdpedigo) [inline graspy/embed/lse.py]: are you against supporting **kwargs here?

File: graspy/embed/mds.py [hunk 3]
```diff
@@ -120,17 +127,20 @@ def fit(self, X):
...
         ----------
-        X : nd-array
+        X : array_like
             If ``dissimilarity=='precomputed'``, the input should be the 
             dissimilarity matrix with shape (n_samples, n_samples). If 
-            ``dissimilarity=='euclidean'``, then the input should be array 
-            with shape (n_samples, n_features) or a nd-array with shape 
-            (n_samples, n_features_1, n_features_2, ..., n_features_d).
+            ``dissimilarity=='euclidean'``, then the input should be 2d-array 
+            with shape (n_samples, n_features) or a 3d-array with shape
```
Comment (bdpedigo) [inline graspy/embed/mds.py]: clarify that the first case (2d) is operating on the nodes, while the second is operating on graphs

File: graspy/embed/svd.py [hunk 4]
```diff
@@ -173,38 +173,96 @@ def select_dimension(X,
...
     -------
-    U: array-like, shape (n_samples, k)
-        the left singular vectors.
-    V: array-like, shape (n_samples, k)
-        the right singular vectors.
-    s: array-like, shape (k)
-        the singular values, as a 1d array.
+    U: array-like, shape (n_samples, n_components)
+        Left singular vectors.
+    D: array-like, shape (n_components)
```
Comment (bdpedigo) [inline graspy/embed/svd.py]: specify that they are returned in non-increasing order of singular values

File: tests/test_svd.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    X_trunc = U_trunc @ np.diag(np.sqrt(D_trunc))
+    norm_trunc = np.linalg.norm(X_trunc, axis=1)
+
+    U_rand, D_rand, V_rand = selectSVD(
+        A, n_components=n_components, algorithm='randomized', n_iter=10)
+    X_rand = U_rand @ np.diag(np.sqrt(D_rand))
+    norm_rand = np.linalg.norm(X_rand, axis=1)
+
+    assert_allclose(norm_full, norm_trunc)
+    assert_allclose(norm_full, norm_rand, rtol=5e-2, atol=5e-2)
```
Comment (bdpedigo) [inline tests/test_svd.py]: how close is allclose for the full/trunc comparison?

File: tests/test_utils.py [hunk 6]
```diff
@@ -47,66 +48,52 @@ def test_to_laplace_IDAD(self):
...
             gus.import_graph(A)
-    
+
     def test_lcc_networkx(self):
-        expected_lcc_matrix = np.array([[0, 1, 1, 0, 0],
-                                        [0, 0, 0, 0, 0],
-                                        [0, 0, 0, 1, 1],
-                                        [0, 1, 0, 0, 0],
+        expected_lcc_matrix = np.array([[0, 1, 1, 0, 0], [0, 0, 0, 0, 0],
+                                        [0, 0, 0, 1, 1], [0, 1, 0, 0, 0],
```
Comment (bdpedigo) [inline tests/test_utils.py]: In several places, why flatten these like this? I think it's harder to read

File: graspy/embed/embed.py (hunk 1 above)
Comment (j1c) [reply graspy/embed/embed.py]: yea i ran into instances where i was calling fit_transform on different data, and realized it wasn't doing anything cause of these lines. you could instantiate more than one class, but seems cumbersome? This is also how fit_transform works in sklearn when I checked.

Comment (bdpedigo) [reply graspy/embed/embed.py]: Yes i agree with this, just wanted to make sure it's the behavior we want

**Pull request merged**